app/services/PersonalDetailsValidationService.scala

```diff
@@ -128,14 +130,12 @@ class PersonalDetailsValidationService @Inject()(connector: PersonalDetailsValid
     }
   }
 
-  def getPDVData(body: PDVRequest)(implicit hc: HeaderCarrier): Future[PDVResponseData] = {
+  def getPDVData(body: PDVRequest)(implicit hc: HeaderCarrier): Future[PDVResponse] = {
     val p = for {
-      pdvData <- createPDVDataFromPDVMatch(body)
-    } yield pdvData match {
-      case data: PDVResponseData =>
-        data
-      case _ =>
-        throw new Exception("No PDV data found")
+      pdv <- createPDVDataFromPDVMatch(body)
+    } yield pdv match {
+      case pdvResponse: PDVResponse => pdvResponse
+      case _ => throw new Exception("No PDV data found")
```

Comment:
This error message is not technically correct as a 404 (not  found) from PDV would be contained within pdvResponse. 

A more suitable error would be 'PDV response could not be parsed'.